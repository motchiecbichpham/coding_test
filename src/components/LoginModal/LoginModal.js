import React, {useState, useEffect} from 'react';
import {
  KeyboardAvoidingView,
  TextInput,
  Text,
  TouchableWithoutFeedback,
  Platform,
  View,
  StyleSheet,
} from 'react-native';
import {COLORS} from '../../utils/colors';
import Button from '../Button/Button';
const LoginModal = ({onClose, onSubmit, error = ''}) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [isDisabled, setIsDisabled] = useState(true);
  useEffect(() => {
    if (password && username) {
      setIsDisabled(false);
    }
  }, [password, username]);
  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      style={styles.keyboardContainer}>
      <TouchableWithoutFeedback onPress={onClose}>
        <View style={styles.container}>
          <View style={styles.content}>
            <Text style={styles.normalText}>Username</Text>
            <TextInput
              style={styles.input}
              placeholder="Enter your username"
              onChangeText={text => {
                setUsername(text);
              }}
              value={username}
              placeholderTextColor={COLORS.silver}
            />
            <Text style={styles.normalText}>Password</Text>
            <TextInput
              style={styles.input}
              onChangeText={text => {
                setPassword(text);
              }}
              placeholder="Enter your password"
              value={password}
              placeholderTextColor={COLORS.silver}
            />
            <View style={styles.buttonsContainer}>
              <Text style={[styles.normalText, {color: COLORS.lightPink}]}>
                {error}
              </Text>
              <Button
                disabled={isDisabled}
                onPress={() => {
                  onSubmit({username: username, password: password});
                }}
                title="Login"
                style={{
                  backgroundColor: isDisabled
                    ? COLORS.gray
                    : COLORS.grayAsparagus,
                }}
              />
            </View>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  keyboardContainer: {
    flex: 1,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.transparent,
  },
  content: {
    width: '80%',
    backgroundColor: COLORS.white,
    borderRadius: 20,
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  input: {
    height: 40,
    borderWidth: 1,
    borderRadius: 8,
    borderColor: COLORS.silver,
    padding: 12,
  },
  normalText: {
    fontWeight: '500',
    marginVertical: 8,
    color: COLORS.darkCharcoal,
  },
  buttonsContainer: {
    marginTop: 12,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
});
export default LoginModal;
