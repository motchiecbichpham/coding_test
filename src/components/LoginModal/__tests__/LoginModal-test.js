import React from 'react';
import renderer from 'react-test-renderer';
import LoginModal from '../LoginModal';

const loginModalProps = {
  onClose: () => {},
  onSubmit: () => {},
};
test('render login modal', () => {
  const tree = renderer.create(<LoginModal {...loginModalProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
