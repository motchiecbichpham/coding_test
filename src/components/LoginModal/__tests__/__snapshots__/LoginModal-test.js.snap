// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login modal 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000080",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 20,
          "paddingHorizontal": 24,
          "paddingVertical": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontWeight": "500",
            "marginVertical": 8,
          }
        }
      >
        Username
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter your username"
        placeholderTextColor="#CACACA"
        style={
          Object {
            "borderColor": "#CACACA",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 40,
            "padding": 12,
          }
        }
        value=""
      />
      <Text
        style={
          Object {
            "color": "#333333",
            "fontWeight": "500",
            "marginVertical": 8,
          }
        }
      >
        Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter your password"
        placeholderTextColor="#CACACA"
        style={
          Object {
            "borderColor": "#CACACA",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 40,
            "padding": 12,
          }
        }
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "space-around",
            "marginTop": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontWeight": "500",
                "marginVertical": 8,
              },
              Object {
                "color": "#EB596E",
              },
            ]
          }
        />
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#808080",
              "borderRadius": 8,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "marginTop": 16,
              "opacity": 1,
              "paddingHorizontal": 16,
              "width": "70%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F2EDD7",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
